java/src/main/kotlin/io/spine/protodata/java/MethodDeclaration.kt

```diff
@@ -0,0 +1,45 @@
+/*
+ * Copyright 2025, TeamDev. All rights reserved.
+ *
+ * Licensed under the Apache License, Version 2.0 (the "License");
+ * you may not use this file except in compliance with the License.
+ * You may obtain a copy of the License at
+ *
+ * https://www.apache.org/licenses/LICENSE-2.0
+ *
+ * Redistribution and use in source and/or binary forms, with or without
+ * modification, must retain the above copyright notice and the following
+ * disclaimer.
+ *
+ * THIS SOFTWARE IS PROVIDED BY THE COPYRIGHT HOLDERS AND CONTRIBUTORS
+ * "AS IS" AND ANY EXPRESS OR IMPLIED WARRANTIES, INCLUDING, BUT NOT
+ * LIMITED TO, THE IMPLIED WARRANTIES OF MERCHANTABILITY AND FITNESS FOR
+ * A PARTICULAR PURPOSE ARE DISCLAIMED. IN NO EVENT SHALL THE COPYRIGHT
+ * OWNER OR CONTRIBUTORS BE LIABLE FOR ANY DIRECT, INDIRECT, INCIDENTAL,
+ * SPECIAL, EXEMPLARY, OR CONSEQUENTIAL DAMAGES (INCLUDING, BUT NOT
+ * LIMITED TO, PROCUREMENT OF SUBSTITUTE GOODS OR SERVICES; LOSS OF USE,
+ * DATA, OR PROFITS; OR BUSINESS INTERRUPTION) HOWEVER CAUSED AND ON ANY
+ * THEORY OF LIABILITY, WHETHER IN CONTRACT, STRICT LIABILITY, OR TORT
+ * (INCLUDING NEGLIGENCE OR OTHERWISE) ARISING IN ANY WAY OUT OF THE USE
+ * OF THIS SOFTWARE, EVEN IF ADVISED OF THE POSSIBILITY OF SUCH DAMAGE.
+ */
+
+package io.spine.protodata.java
+
+/**
+ * A declaration of a Java method.
+ *
+ * An example usage:
+ *
+ * ```
+ * val validate = MethodDeclaration(
+ *     """
+ *     public java.util.Optional<io.spine.validate.ValidationError> validate() {
+ *         var noParent = io.spine.base.FieldPath.getDefaultInstance();
+ *         return validate(noParent);
+ *     }
+ *     """.trimIndent()
```

Comment:
What is the outcome of your investigation regarding `trimIndent()` and `trimMargin()`?

Reply:
PSI throws when the passed method declaration has leading whitespaces. I've left a [note](https://github.com/SpineEventEngine/validation/pull/188/files#diff-f2902148e4d7730176b75979fb22fb1e06e360d096b01eb46dd6af020e7dbac8R190), and have used the simplest `trim()`.

Reply:
@yevhenii-nadtochii The question is boiled down to this: will this "example usage" work?